tests/test_chiral_restraints.py

```diff
@@ -626,5 +635,30 @@ def assert_same_signs(a, b):
         frame_bond_vols = get_chiral_bond_volumes(f)
         # f = f - np.mean(f, axis=0)
         # writer.write_frame(f*10)
-        assert_same_signs(frame_atom_vols, ref_chiral_atom_vols)
-        assert_same_signs(frame_bond_vols, ref_chiral_bond_vols)
+        if check_chiral_atoms:
+            assert_same_signs(frame_atom_vols, ref_chiral_atom_vols)
+        if check_chiral_bonds:
+            assert_same_signs(frame_bond_vols, ref_chiral_bond_vols)
+
+
+def test_chiral_inversion_in_single_topology():
+    very_short_hrex_params = replace(
+        DEFAULT_HREX_PARAMS, n_frames=2, n_eq_steps=2, steps_per_frame=2, hrex_params=HREXParams(n_frames_bisection=2)
+    )
+    ff = Forcefield.load_default()
+
+    # mol_a, mol_b : substituted chiral cyclobutyl
+    # with 2 different alignments of mol_b w.r.t. mol_a
+    mol_dict = {utils.get_mol_name(mol): mol for mol in utils.read_sdf("tests/data/1243_chiral_ring_confs.sdf")}
+    mol_a = mol_dict["A"]
+    mol_b_0 = mol_dict["B_0"]
+    mol_b_1 = mol_dict["B_1"]
+    core_0 = get_cores(mol_a, mol_b_0, **DEFAULT_ATOM_MAPPING_KWARGS)[0]
+    core_1 = get_cores(mol_a, mol_b_1, **DEFAULT_ATOM_MAPPING_KWARGS)[0]
+    assert len(core_0) == 10
+    assert len(core_1) == 15
+
+    _ = run_vacuum(mol_a, mol_b_0, core_0, ff, None, very_short_hrex_params, n_windows=3)
```

Comment:
What is `run_vacuum` used here for? It looks like you don't rely on any output from this function call. Is it to verify no exception is triggered, if so could we target the code specifically that we might expect to raise the assertion rather then run vacuum HREX?

Reply:
Oh, looks like this test is still incomplete

Reply:
>Is it to verify no exception is triggered, if so could we target the code specifically that we might expect to raise the assertion rather then run vacuum HREX?

Correct -- all this does is run ~2 steps of MD in 3 windows, hopefully without throwing an exception. The main places I wanted to check that might raise an exception would be: (1) invoking the single topology constructor, (2) constructing chiral atom GPU potentials / MD contexts, (3) producing diagnostics that depend on list of potentials.

Reply:
It would be good to add a comment with the detail you added, makes perfect sense with that context.